src/coprocessor/dag/expr/compare.rs

```diff
@@ -380,12 +544,21 @@ fn like(target: &[u8], pattern: &[u8], escape: u32, recurse_level: usize) -> Res
     }
 }
 
+fn regexp(target: &str, pattern: &str) -> Result<Option<i64>> {
```

Comment:
Seems this function would never return `Ok(None)`? Maybe we could return `Result<i64>` directly, and use `regexp().map(Some)` as following:

```
fn regexp(target: &str, pattern: &str) -> Result<i64> {
    Ok(Regex::new(pattern)?.is_match(target) as i64)
}
```
even we do not need to define this function since it's too easy?